Sources/JWTKeychain/Controllers/API/UserController.swift

```diff
@@ -66,13 +72,109 @@ open class UserController: UserControllerType {
 
     open func regenerate(request: Request) throws -> ResponseRepresentable {
         let user = try request.user()
-        let token = try self.configuration.generateToken(userId: user.id!)
+        let token = try self.configuration.generateToken(user: user)
         return try JSON(node: ["token": token])
     }
 
     open func me(request: Request) throws -> ResponseRepresentable {
         let user = try request.user()
-        let token = try self.configuration.generateToken(userId: user.id!)
+        let token = try self.configuration.generateToken(user: user)
         return try user.makeJSON(token: token)
     }
+
+    open func resetPasswordEmail(request: Request) -> ResponseRepresentable {
+        do {
+
+            guard
+                let email = request.data["email"]?.string,
+                let user: User = try User.query().filter("email", email).first() else {
+                    return JSON(["success": "Instructions were sent to the provided email"])
+            }
+
+            let token = try self.configuration.generateResetPasswordToken(user: user)
+
+            // Send mail
+            try Mailer.sendResetPasswordMail(drop: self.drop, user: user, token: token)
```

Comment:
@BrettRToomey mentioned this issue https://github.com/vapor/vapor/issues/781 which discusses the idea of creating a generail mail-sending protocol (which makes a lot of sense to me). Until we get this into Vapor, maybe we should just make our own simple temporary protocol and make `Mailer` conform to it - and then inject this into this controller.